### DISCUSSION TITLE
File scoped class

### DISCUSSION THREAD

**Discussion by vnesterovsky** [dotnet/csharplang#5785] [Language Ideas] [answered]
Similarly to C#'s 10 file scoped namespace, which allows to remove redundant indentations, it might be a reasonal to have file scope classes.

Arguments are the same, in many cases file have a single class, containing members.
As a syntactic sugar I suggest to consider a syntax.

```C#
namespace X.Y.Z;

using System;

class X;

// members...
public X() {...}

public int Property { get; set; }

private var x = 7; // a field.
```

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/discussions/928

**Reply by vnesterovsky** [reply]
Thank you.
I missed that proposal.
